packages/client/src/components/ToolDetailsPage/ToolDetailsPage.component.js

```diff
@@ -0,0 +1,124 @@
+import React, { useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+import getApiBaseUrl from '../../utils/getApiBaseURL';
+import './ToolDetailsPage.css';
+
+export const ToolDetailsPage = () => {
+  const [tool, setTool] = useState({});
+  const [isLoading, setIsLoading] = useState(true);
+  const [isDataEmpty, setisDataEmpty] = useState(false);
+  const params = useParams();
+
+  useEffect(() => {
+    async function fetchTool(id) {
+      const response = await fetch(`${getApiBaseUrl()}/api/tools/${id}`);
+      const result = await response.json();
+      return result;
+    }
+    function setFetchTool() {
+      setIsLoading(true);
+
+      fetchTool(params.id)
+        .then((data) => {
+          if (typeof data === 'object') {
+            setisDataEmpty(false);
+            const toolWithCategries = data[0];
+            toolWithCategries.categories = data.categories;
+            setTool(toolWithCategries);
```

Comment:
If you change the backend to get the categories right away, then the lines above will not be needed 🙏🏽